src/System.Security.Cryptography.Algorithms/src/Internal/Cryptography/AppleCCCryptor.cs

```diff
@@ -0,0 +1,248 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Diagnostics;
+using System.Security.Cryptography;
+
+namespace Internal.Cryptography
+{
+    internal sealed class AppleCCCryptor : BasicSymmetricCipher
+    {
+        private readonly bool _encrypting;
+        private SafeAppleCryptorHandle _cryptor;
+
+        public AppleCCCryptor(
+            Interop.AppleCrypto.PAL_SymmetricAlgorithm algorithm,
+            CipherMode cipherMode,
+            int blockSizeInBytes,
+            byte[] key,
+            byte[] iv,
+            bool encrypting)
+            : base(cipherMode.GetCipherIv(iv), blockSizeInBytes)
+        {
+            _encrypting = encrypting;
+
+            OpenCryptor(algorithm, cipherMode, key);
+        }
+
+        protected override void Dispose(bool disposing)
+        {
+            if (disposing)
+            {
+                _cryptor?.Dispose();
+                _cryptor = null;
+            }
+
+            base.Dispose(disposing);
+        }
+
+        public override int Transform(byte[] input, int inputOffset, int count, byte[] output, int outputOffset)
+        {
+            Debug.Assert(input != null);
+            Debug.Assert(inputOffset >= 0);
+            Debug.Assert(count > 0);
+            Debug.Assert((count % BlockSizeInBytes) == 0);
+            Debug.Assert(input.Length - inputOffset >= count);
+            Debug.Assert(output != null);
+            Debug.Assert(outputOffset >= 0);
+            Debug.Assert(output.Length - outputOffset >= count);
```

Comment:
Do we have line numbers in stack traces yet on OSX?  Especially if not, consider adding messages for all of these asserts, to help determine which one failed if any fails.  Doing so would also let you include the value that caused the failure, e.g. ```Debug.Assert(inputOffset >= 0, $"Expected non-negative inputOffset, got {inputOffset}");```